src/lsm/lsm_merge.c

```diff
@@ -235,6 +235,14 @@ __lsm_merge_span(WT_SESSION_IMPL *session, WT_LSM_TREE *lsm_tree,
 			break;
 
 		/*
+		 * If we have enough chunks for a merge and the next chunk is
+		 * in too high a generation, stop.
+		 */
+		if (nchunks >= merge_min &&
+		    chunk->generation - youngest_gen > max_gap)
```

Comment:
I don't think it's safe to do this comparison until `youngest_gen` has been updated  - otherwise there is a possibility that `chunk->generation` is smaller than `youngest_gen` and there is an underflow.

Reply:
eee yes, an underflow bug would be introduced 👍 
By the point of this predicate it's guaranteed that ```old_gen - yon_gen <= max_gen```, otherwise it couldn't reach here. Therefore, if this predicate hits and jumps out, we'll get an valid start & end pair.
However, if we place this predicate after  ```youngest_gen & oldest_gen``` is updated then break out,
it's possible that ``` old_gen - yon_gen > max_gen ``` and will fall into [code](https://github.com/squalfof/wiredtiger/blob/f3770bf512d40a3071d402c39d0faaac4a0ec1ea/src/lsm/lsm_merge.c#L306), an valid start & end will be ignored.
how about keep this predication here with ?

``` (int32_t)chunk->generation - youngest_gen > (int32_t)max_gap```

Reply:
Your suggestion should be correct, but we try to avoid casting in our code base. How about changing it to:

```
chunk->generation > youngest_gen + max_gap
```